src/gc/gc.d

```diff
@@ -3250,6 +3250,23 @@ unittest // bugzilla 14467
     assert(arr.capacity);
 }
 
+unittest // bugzilla 15353
+{
+    import core.memory : GC;
+
+    static struct Foo
+    {
+        ~this()
+        {
+            GC.free(buf); // ignored in finalizer
+        }
+
+        void* buf;
+    }
+    new Foo(GC.malloc(10));
```

Comment:
This should probably be in a loop, to make sure there is no left-over pointer in the registers or stack.